.github/workflows/build-ubuntu.yml

```diff
@@ -0,0 +1,272 @@
+                e2eTests: false,
+                checkCodeFormat: false,
+              },
+              {
+                os: ubuntu-20.04,
+                cmakeBuildType: Release,
+                asanEnabled: false,
+                guiEnabled: false,
+                cudaEnabled: true,
+                e2eTests: false,
+                checkCodeFormat: false,
+              },
+        ]
+
+    env:
+      COMPILER_CACHE_VERSION: 1
+      COMPILER_CACHE_DIR: ${{ github.workspace }}/compiler-cache
+      CCACHE_DIR: ${{ github.workspace }}/compiler-cache/ccache
+      CCACHE_BASEDIR: ${{ github.workspace }}
+      CTCACHE_DIR: ${{ github.workspace }}/compiler-cache/ctcache
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/cache@v4
+        id: cache-builds
+        with:
+          key: v${{ env.COMPILER_CACHE_VERSION }}-${{ matrix.config.os }}-${{ matrix.config.arch }}-${{ matrix.config.cmakeBuildType }}-${{ matrix.config.asanEnabled }}--${{ matrix.config.cudaEnabled }}-${{ github.run_id }}-${{ github.run_number }}
+          restore-keys: v${{ env.COMPILER_CACHE_VERSION }}-${{ matrix.config.os }}-${{ matrix.config.arch }}-${{ matrix.config.cmakeBuildType }}-${{ matrix.config.asanEnabled }}--${{ matrix.config.cudaEnabled }}
+          path: ${{ env.COMPILER_CACHE_DIR }}
+      
+      - name: Install compiler cache
+        run: |
+          mkdir -p "$CCACHE_DIR" "$CTCACHE_DIR"
+          echo "$COMPILER_CACHE_DIR/bin" >> $GITHUB_PATH
+
+          if [ -f "$COMPILER_CACHE_DIR/bin/ccache" ]; then
+            exit 0
+          fi
+
+          set -x
+          wget https://github.com/ccache/ccache/releases/download/v4.8.2/ccache-4.8.2-linux-x86_64.tar.xz
+          echo "0b33f39766fe9db67f40418aed6a5b3d7b2f4f7fab025a8213264b77a2d0e1b1  ccache-4.8.2-linux-x86_64.tar.xz" | sha256sum --check
+          tar xfv ccache-4.8.2-linux-x86_64.tar.xz
+          mkdir -p "$COMPILER_CACHE_DIR/bin"
+          mv ./ccache-4.8.2-linux-x86_64/ccache "$COMPILER_CACHE_DIR/bin"
+          ctcache_commit_id="66c3614175fc650591488519333c411b2eac15a3"
+          wget https://github.com/matus-chochlik/ctcache/archive/${ctcache_commit_id}.zip
+          echo "108b087f156a9fe7da0c796de1ef73f5855d2a33a27983769ea39061359a40fc  ${ctcache_commit_id}.zip" | sha256sum --check
+          unzip "${ctcache_commit_id}.zip"
+          mv ctcache-${ctcache_commit_id}/clang-tidy* "$COMPILER_CACHE_DIR/bin"
+
+      - name: Check code format
+        run: |
+          if [ "${{ matrix.config.checkCodeFormat }}" != "true" ]; then
+            exit 0
+          fi
+          set +x -euo pipefail
+          sudo apt-get update && sudo apt-get install -y clang-format-14 black
+          ./scripts/format/clang_format.sh
+          ./scripts/format/black.sh
+          git diff --name-only | xargs -r printf "##vso[task.LogIssue type=error;sourcepath=%s;linenumber=1;columnnumber=1;]Invalid code format, run scripts/format/*.sh\n"
```

Comment:
This logging syntax seems to be ADO-specific. In GH actions the equivalent is described here: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message

Reply:
Thx, good catch.